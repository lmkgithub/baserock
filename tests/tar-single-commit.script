#!/bin/sh
#
# Tests converting a simple SVN repository to git.
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -e

logfile="$DATADIR/svn-test-repo.log"
workdir="$DATADIR/work-dir"

./lorry --pull-only --log="$logfile" --working-area="$workdir" \
  "$DATADIR/tar-test-repo.lorry" > /dev/null 2> /dev/null

# verify that the git repositories were created successfully
for FORMAT in "gzip" "bzip2" "lzma"; do
  cd "$workdir/tar-test-repo-$FORMAT/git/"

  echo "$FORMAT"

  # list the branches
  git show-ref | cut -d' ' -f2

  # cat the test file
  git cat-file blob master:test.txt

  # list the commit messages
  git log --pretty='%s' master
done
