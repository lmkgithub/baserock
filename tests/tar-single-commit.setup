#!/bin/sh
#
# Creates gzip/bzip2/lzma tarballs, each with a single file.
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -e

# create the original "repository"
repo="$DATADIR/tar-test-repo"
mkdir "$repo"
echo "first line" > "$repo/test.txt"

# create the tarballs
cd "$DATADIR"
tar -czf tar-test-repo.tar.gz "`basename $repo`"
tar -cjf tar-test-repo.tar.bz2 "`basename $repo`"
tar -cf tar-test-repo.tar.lzma "`basename $repo`" --lzma

# create the .lorry file for the tarball "repositories"
cat <<EOF > $DATADIR/tar-test-repo.lorry
{
  "tar-test-repo-gzip": {
    "type": "tarball",
    "compression": "gzip",
    "strip": 1,
    "url": "file://$DATADIR/tar-test-repo.tar.gz"
  },
  "tar-test-repo-bzip2": {
    "type": "tarball",
    "compression": "bzip2",
    "strip": 1,
    "url": "file://$DATADIR/tar-test-repo.tar.bz2"
  },
  "tar-test-repo-lzma": {
    "type": "tarball",
    "compression": "lzma",
    "strip": 1,
    "url": "file://$DATADIR/tar-test-repo.tar.lzma"
  }
}
EOF

# create the working directory
test -d "$DATADIR/work-dir" || mkdir "$DATADIR/work-dir"
