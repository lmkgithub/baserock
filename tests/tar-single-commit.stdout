gzip
refs/heads/master
first line
Tarball conversion
bzip2
refs/heads/master
first line
Tarball conversion
lzma
refs/heads/master
first line
Tarball conversion
